### PR TITLE
Limit layers extent to the extent of the portal

### PR DESCRIPTION
**Pull request by Jenselme**
It prevents to load tile that are outside the scope of the portal. It
saves bandwidth, reduces load time, avoids to display unuseful
information.

### CODE REVIEW

File: src/components/map/MapService.js [hunk 1]
```diff
@@ -1308,6 +1313,18 @@ goog.require('ga_urlutils_service');
             easing: ol.easing.easeOut
           }));
           map.getView().setRotation(0);
+        },
+
+        adaptExtentPortalExtent: function(extent) {
```
Comment (gjn) [inline src/components/map/MapService.js]: Maybe adapt function name to reflect better what it does? Like `mergeWithDefaultExtent`.

**Comment by gjn** [comment]
This is nice. Please rebase and probably adapt a function name.

**Comment by Jenselme** [comment]
- rebased
- rename `adaptExtentPortalExtent` into `mergeWithDefaultExtent`

**Comment by gjn** [comment]
Thanks alot for this!

File: src/components/map/MapService.js [hunk 2]
```diff
@@ -375,7 +376,7 @@ goog.require('ga_urlutils_service');
           var wmsOptions = {
             url: getCapLayer.wmsUrl,
             label: getCapLayer.Title,
-            extent: getCapLayer.extent,
+            extent: gaMapUtils.mergeWithDefaultExtent(getCapLayer.extent),
```
Comment (oterral) [inline src/components/map/MapService.js]: Hmm I'm not sure of this one, if the layer's extent define by the GetCap is 10km aroud Zurich for example. I don't think we want the layer was requested outside this extent

Comment (oterral) [inline src/components/map/MapService.js]: Forget this I hadn't  read what the function does

Comment (gjn) [inline src/components/map/MapService.js]: AFAIK, the mergeWith is actually an intersectWith, @Jenselme ?

Comment (Jenselme) [inline src/components/map/MapService.js]: From what I understand of your comment, this line solves this.

Comment (Jenselme) [inline src/components/map/MapService.js]: > AFAIK, the mergeWith is actually an intersectWith, @Jenselme ?

Indeed, this name looks better. Should I open a new PR?

Comment (gjn) [inline src/components/map/MapService.js]: I would merge it in a second :=)

**Pull request merged**